### PR TITLE
Fix version query param on ui api collection detail endpoint

### PR DESCRIPTION
**Pull request by awcrosby**
Supports version dropdown on collection detail page.

https://issues.redhat.com/browse/AAH-90

Issue: AAH-90

### CODE REVIEW

File: galaxy_ng/app/api/ui/viewsets/collection.py [hunk 1]
```diff
@@ -24,35 +25,41 @@
...
-    # https://pulp.plan.io/issues/7428
-    def get_queryset(self):
-        """
-        Overrides to not use is_highest param and
-        ensures one collection version per collection is returned.
-        """
-        distro_content = self.get_distro_content(self.kwargs["path"])
+    def get_object(self):
+        """Return CollectionVersion object, latest or via query param 'version'."""
+        version = self.request.query_params.get('version', None)
```
Comment (alikins) [inline galaxy_ng/app/api/ui/viewsets/collection.py:47]: Would this be a reasonable place to set a 'lookup_field' on the viewset (ala, https://www.django-rest-framework.org/api-guide/generic-views/#attributes)

Comment (alikins) [reply galaxy_ng/app/api/ui/viewsets/collection.py:47]: hmm, nm. Given all the other cases to handle, 'lookup_field' isn't super useful

File: galaxy_ng/app/api/ui/serializers/collection.py [hunk 2]
```diff
@@ -121,31 +121,21 @@ def get_namespace(self, obj):
     def get_deprecated(self, obj):
         return obj.collection.deprecated
 
-    # TODO(awcrosby): refactor once pulp_ansible is_highest param filters by repo
-    # https://pulp.plan.io/issues/7428
     def _get_versions_in_repo(self, obj):
```
Comment (alikins) [inline galaxy_ng/app/api/ui/serializers/collection.py:124]: Feels little odd to define _get_versions_in_repo in this class, but only use it in CollectionDetailSerializer.
But shrug, _CollectionSerializer is always subclass so doesn't matter a ton.

File: galaxy_ng/app/api/ui/viewsets/collection.py (hunk 1 above)
Comment (alikins) [inline galaxy_ng/app/api/ui/viewsets/collection.py:47]: Wondering if some of the logic here would fit better in CollectionViewSet.retrieve() ?

Comment (alikins) [reply galaxy_ng/app/api/ui/viewsets/collection.py:47]: Though, I guess this needs to apply to retrieve() and list(), so get_object() is probably correct.

**Review by alikins — APPROVED** [review]
Haven't tested, but LGTM

**Review by newswangerd — APPROVED** [review]

File: galaxy_ng/app/api/ui/serializers/collection.py [hunk 3]
```diff
@@ -155,6 +145,7 @@ def get_all_versions(self, obj):
         versions_in_repo = self._get_versions_in_repo(obj)
         return CollectionVersionSummarySerializer(versions_in_repo, many=True).data
 
+    # TODO(awcrosby): rename field to "version_details" since with
+    # "version" query param this won't always be the latest version
```
Comment (alikins) [inline galaxy_ng/app/api/ui/serializers/collection.py:149]: Or possibly in the future, just always return a list for 'version' and let clients assume the first item is the highest or best version.

**Pull request merged**